Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -567,6 +567,15 @@ WebIDL::ExceptionOr<void> Document::run_the_document_write_steps(StringView inpu
 // https://html.spec.whatwg.org/multipage/dynamic-markup-insertion.html#dom-document-open
 WebIDL::ExceptionOr<Document*> Document::open(Optional<String> const&, Optional<String> const&)
 {
+    // If document belongs to a child navigable, we need to make sure its initial navigation is done,
+    // because subsequent steps will modify "initial about:blank" to false, which would cause
+    // initial navigation to fail in case it was "about:blank".
+    if (navigable() && navigable()->container()) {
+        HTML::main_thread_event_loop().spin_until([&] {
+            return navigable()->container()->content_navigable_initialized();
+        });
+    }
```

Comment:
We know `navigable()` is very heavy to call, so let's cache it in a local variable.

Reply:
This doesn't feel right. What if `spin_until()` causes some other JS execution to happen?

Reply:
I replaced it with `spin_processing_tasks_with_source_until()` to execute only tasks from navigation source. it should not be interleaved with any JS execution now.